test/linalg/symmetric.jl

```diff
@@ -42,206 +41,229 @@ end
+                    @test ctranspose(S) ==  Symmetric(conj(asym))
+                    @test  transpose(H) ==  Hermitian(transpose(asym))
+                    @test ctranspose(H) === H == asym
+                end
+            end
         end
 
-        #inversion
-        @test inv(Hermitian(asym)) ≈ inv(asym)
-        if eltya <: Real && eltya != Int
-            @test inv(Symmetric(asym)) ≈ inv(asym)
-            @test inv(Hermitian(a)) ≈ inv(full(Hermitian(a)))
-            @test inv(Symmetric(a)) ≈ inv(full(Symmetric(a)))
-        end
+        @testset "linalg unary ops" begin
+            # Revisit when implemented in julia
+            if eltya != BigFloat
+                @testset "trace" begin
+                    @test trace(asym) == trace(Hermitian(asym))
+                end
+                @testset "cond" begin
+                    @test cond(Hermitian(asym)) ≈ cond(asym)
+                end
+                
+                @testset "inversion" begin
+                    @test inv(Hermitian(asym)) ≈ inv(asym)
+                    if eltya <: Real && eltya != Int
+                        @test inv(Symmetric(asym)) ≈ inv(asym)
+                        @test inv(Hermitian(a)) ≈ inv(full(Hermitian(a)))
+                        @test inv(Symmetric(a)) ≈ inv(full(Symmetric(a)))
+                    end
+                end
+
+                @testset "symmetric eigendecomposition" begin
+                    d, v = eig(asym)
+                    @test asym*v[:,1] ≈ d[1]*v[:,1]
+                    @test v*Diagonal(d)*v' ≈ asym
+                    @test isequal(eigvals(asym[1]), eigvals(asym[1:1,1:1]))
+                    @test abs.(eigfact(Hermitian(asym), 1:2)[:vectors]'v[:,1:2]) ≈ eye(eltya, 2)
+                    eig(Hermitian(asym), 1:2) # same result, but checks that method works
+                    @test abs.(eigfact(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2)[:vectors]'v[:,1:2]) ≈ eye(eltya, 2)
+                    eig(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2) # same result, but checks that method works
+                    @test eigvals(Hermitian(asym), 1:2) ≈ d[1:2]
+                    @test eigvals(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2) ≈ d[1:2]
+                    @test full(eigfact(asym)) ≈ asym
+                    @test eigvecs(Hermitian(asym)) ≈ eigvecs(asym)
+
+                    # relation to svdvals
+                    @test sum(sort(abs.(eigvals(Hermitian(asym))))) == sum(sort(svdvals(Hermitian(asym))))
+                end
+                
+                @testset "det" begin
+                    @test det(asym) ≈ det(Hermitian(asym, :U))
+                    @test det(asym) ≈ det(Hermitian(asym, :L))
+                    if eltya <: Real
+                        @test det(asym) ≈ det(Symmetric(asym, :U))
+                        @test det(asym) ≈ det(Symmetric(asym, :L))
+                    end
+                    @test det(a + a.') ≈ det(Symmetric(a + a.', :U))
```

Comment:
This is now `asym` so this and test on line below are the same as the two tests in the `if`.